src/axolotl/prompters.py

```diff
@@ -187,18 +231,18 @@ class Conversation:
     offset: int
     sep_style: SeparatorStyle = SeparatorStyle.SINGLE
     sep: str = "###"
-    sep2: str = None
+    sep2: Optional[str] = None
 
     def get_prompt(self) -> Generator[str, None, None]:
-        seps = [self.sep, self.sep2]
-        preamble = self.system + seps[0]
+        # seps = [self.sep, self.sep2]
+        preamble = self.system + self.sep
         yield preamble
-        for i, (role, message) in enumerate(self.messages):
+        for _, (role, message) in enumerate(self.messages):
             if message:
-                yield (role + ":", " " + message)
+                yield role + ":" + " " + message
             else:
-                logging.warning("role with empty message: " + role)
-                yield (role + ":",)
+                logging.warning(f"role with empty message: {role}")
+                yield role + ":"
```

Comment:
To satisfy `mypy`, I had to change from both `yield (str, str)` to just `yield str`. I am not sure if this broke anything. The only place that seem to call this seems to expect one value only.

Reply:
so I think `get_prompt` needs to return `Generator[Tuple[str, Optional[str]], None, None]` 
(or maybe `Generator[Union[str, Tuple[str, Optional[str]]], None, None]` ? )
then `ShareGPTPrompter.build_prompt` also needs to return `Generator[Tuple[str, Optional[str]], None, None]`, because if you look at `ShareGPTPromptTokenizingStrategy.tokenize_prompt` , it calls:
```
            for i, part in enumerate(
                self.prompter.build_prompt(self.get_conversation_thread(prompt))
            ):
```
it expects either a Tuple or str 
```
if isinstance(part, tuple):
    ...
else:
    ...
```